diff --git a/bam.lua b/bam.lua
index 77a1aab7..0c49fcdc 100644
--- a/bam.lua
+++ b/bam.lua
@@ -177,6 +177,8 @@ function GenerateLinuxSettings(settings, conf, arch, compiler)
 		settings.link.flags:Add("-m64")
 	elseif arch == "armv7l" then
 		-- arm 32 bit
+	elseif arch == "aarch64" then
+		-- arm 64 bit
 	else
 		print("Unknown Architecture '" .. arch .. "'. Supported: x86, x86_64")
 		os.exit(1)
diff --git a/src/base/detect.h b/src/base/detect.h
index 50931623..ddfc4e0f 100644
--- a/src/base/detect.h
+++ b/src/base/detect.h
@@ -142,13 +142,21 @@
 
 #if defined(__arm__)
 	#define CONF_ARCH_ARM 1
-	#define CONF_ARCH_STRING "arm"
+	#define CONF_ARCH_STRING "armx"
 	#if !defined(CONF_ARCH_ENDIAN_LITTLE) && !defined(CONF_ARCH_ENDIAN_BIG)
 		#define CONF_ARCH_ENDIAN_LITTLE 1
 	#endif
 #endif
 
 
+#if defined(__aarch64__)
+	#define CONF_ARCH_AARCH64 1
+	#define CONF_ARCH_STRING "aarch64"
+	#if !defined(CONF_ARCH_ENDIAN_LITTLE) && !defined(CONF_ARCH_ENDIAN_BIG)
+		#define CONF_ARCH_ENDIAN_LITTLE 1
+	#endif
+#endif
+
 #ifndef CONF_FAMILY_STRING
 #define CONF_FAMILY_STRING "unknown"
 #endif
